Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -7844,6 +7844,18 @@ void WebPageProxy::beginSafeBrowsingCheck(const URL&, API::Navigation&, bool for
 }
 #endif
 
+void WebPageProxy::decidePolicyForNavigationActionAndGetBackForwardListAsync(IPC::Connection& connection, NavigationActionData&& data, FrameIdentifier frameID, CompletionHandler<void(PolicyDecision&&, Vector<Ref<FrameState>>&&)>&& completionHandler)
```

Comment:
Seems like this should just outright replace `decidePolicyForNavigationActionAsync` but I presume you separated this out to make it easier to iterate on changes?